src/operations/message_op.h

```diff
@@ -0,0 +1,171 @@
+/*******************************************************************************
+*   Ark Wallet
+*   (c) 2017 Ledger
+*   (c) ARK Ecosystem
+*
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+********************************************************************************/
+
+#ifndef ARK_OPERATIONS_MESSAGE_H
+#define ARK_OPERATIONS_MESSAGE_H
+
+#include <stdint.h>
+
+#include <os.h>
+
+#include "constants.h"
+
+#include "operations/status.h"
+
+#include "utils/hex.h"
+#include "utils/print.h"
+
+#include "ux.h"
+#include "ux/display_context.h"
+
+////////////////////////////////////////////////////////////////////////////////
+
+// Sets the titles and variables to display a Message.
+//
+// Message must be: 0 < Msg <= 255
+//
+// -------
+// Display:
+//
+// Message > 0 && Message <= 64
+// - screen 1: Message Length
+// - screen 2: \"message\"
+//
+// ---
+//
+// Message > 64 && Message <= 128
+// - screen 1: Message Length
+// - screen 2: \"message...\"
+// - screen 3: \"...message\"
+//
+// ---
+//
+// Message > 128 && Message <= 192
+// - screen 1: Message Length
+// - screen 2: \"message...\"
+// - screen 3: \"...message...\"
+// - screen 4: \"...message\"
+//
+// ---
+//
+// Message > 192 && Message <= 255
+// - screen 1: Message Length
+// - screen 2: \"message...\"
+// - screen 3: \"...message...\"
+// - screen 4: \"...message...\"
+// - screen 5: \"...message\"
+//
+// ---
+static void internalHandleMessage(const uint8_t *buffer, const uint32_t length) {
+    if (length == 0 || length > MAX_DISPLAY_BUFFER) {
+        THROW(0x6A80);
+    }
+
+    // Set the operation title.
+    os_memmove((char *)displayCtx.operation, "Message", 8U);
+
+    // Set the Message Length for display.
+    os_memmove((char *)displayCtx.title[0], "length:", 8U);
+
+    printAmount(length,
+                displayCtx.var[0], sizeof(displayCtx.var[0]),
+                "", 0U, 0U);
+
+    // Set the first part of the Message.
+    os_memmove((char *)displayCtx.title[1], "message:", 9U);
+    os_memmove((char *)displayCtx.var[1],
+                buffer,
+                MIN(length, HASH_64_LENGTH));
```

Comment:
* Put "..." into a variable:
```
const char* ellipses = "...";
const size_t ellipses_len = strlen(ellipses);
```
* Use `ellipses` instead of `"..."`
* Use `ellipses_len` instead of `3` and `ellipses_len + 1` instead of `4`.
* Replace `HASH_64_LENGTH` with `const size_t maxUseful = sizeof(displayCtx.var[0]) - 2 * strlen(ellipses) - 1 /* terminating '\0' */;` - it cut 65 chars instead of 64, but more importantly, it would hint the reader of the code where the limit comes from.